invenio_records_resources/services/records/search.py

```diff
@@ -21,3 +24,49 @@ def terms_filter(field):
     def inner(values):
         return Q('terms', **{field: values})
     return inner
+
+
+def nested_terms_filter(field, subtype_field, splitchar):
```

Comment:
Minor: Can you rename occurrences of subtype -> subfield:

```suggestion
def nested_terms_filter(field, subfield, splitchar):
```